Jenkinsfile

```diff
@@ -555,10 +557,12 @@ pipeline {
                     }
                     agent {
                         dockerfile {
-                            filename 'utils/docker/Dockerfile.centos.7'
+                            filename 'Dockerfile.checkpatch'
+                            dir 'utils/docker'
                             label 'docker_runner'
-                            additionalBuildArgs dockerBuildArgs() +
-                                           " -t ${sanitized_JOB_NAME}-centos7 "
+                            additionalBuildArgs dockerBuildArgs(cachebust:false,
+                                                                add_repos:false) +
+                                                " --build-arg CACHEBUST=" + current_time.get(Calendar.WEEK_OF_YEAR)
```

Comment:
What about security updates?  Are we going to potentially live with an image that could have a security vulnerability in it for up to a week?

Reply:
Yes, that's what I'm proposing.  How often do we update Jenkins itself?

Reply:
Weekly, as needed (there isn't a new version out every week I don't think).  But Jenkins is entirely different as it disrupts the entire team to update it.

Reply:
How often do you think would be appropriate and do we have an internal up-to-date fedora:31 mirror?  Going outside the network for every PR seems overkill and unreliable.

How often does our mirror update for the repos that we do store cached versions of?

Reply:
> How often do you think would be appropriate

For security purposes, probably at least daily.

> and do we have an internal up-to-date fedora:31 mirror?

Surely you meant `fedora:latest` which is `fedora:33` at the moment, and yes, our caching proxy server will always be as up-to-date as the clients that update through it.

> Going outside the network for every PR seems overkill and unreliable.

Yes, we should always be using our internal mirror/caching-proxy-server.  That means having to make our Dockerfiles live both inside of our network with the luxury of our mirror as well as outside using regular upstream mirrors.  There are examples of how we do this in the Dockerfile.centos.7 file.
 
> How often does our mirror update for the repos that we do store cached versions of?

The mirror is constantly being updated as clients update through it.

Reply:
I will point out, though this conversation is a bit older now, there is also an in between I think, where it could update weekly, and we could just register for PSIRT notifications for the rare event there is something that can't wait a week, if it's going to cause challenges to our systems to handle keeping them that up to date (whether due to brittleness, or load, or complexity, or just in general making things take longer on a regular basis). These things may not be as an obvious a disruption to the team as a Jenkins update, but can be impactful all the same...I hope we are weighing those things too.

Reply:
> we could just register for PSIRT notifications for the rare event there is something that can't wait a week

I'm not sure what that means really, what granularity it's at, etc.  If it's something that needs action every time a new O/S and/or release/version is brought into our CI pipeline, that sounds like something that is going to fail to continue to get updated.  There is just too much of the left hand not knowing what the right hand is doing around here with many people updating Dockerfiles, CI test nodes, builds, etc.

On the other hand if it's a notification about vulnerabilities in every release of every O/S out there, that sounds like a haystack of notifications and somebody has to go through all of our uses of O/S releases to see what's vulnerable and what's not.

And then there is the case where we are updating say, weekly and a PSIRT does come out.  That becomes one of those very disruptive _everyone drop everything_ to fix it situations that chews up a few man-days.

These are all of the things I am trying to avoid.  Updating frequently, I don't think is unachievable.  We have lots of caching around here that should hopefully combat brittleness.  And we currently already update most of our images every time we use them (so as frequently as is actually possible) and that seems to be working fine, other than the update time gets longish when base images are not updated frequently enough.  But that's a solvable problem, with resources to apply to it.

Reply:
I could be wrong, but, I thought it had some granularity to what projects/products you could subscribe to. It certainly would require someone to own following up on them, but since we already need to get those for our dependencies to get notified of CVEs, I would think we could just roll that in to the same ownership/checking. And I don't think that having to update suddenly when it's updating weekly already is likely to be a very onerous task, no? Not like there's likely to be lots of regressions to yak-shave, etc. So while unexpected, I would think they'd be more likely to be a pretty simple activity, no?

And not necessarily saying we shouldn't automate these updates, just want to make sure we're also weighing those pros/cons with other options. Longish updates are one of those cons that are rather impactful. It is solvable, yes, but not solved yet. So perhaps there is an ordering of prioritizing these improvements that can minimize impact to our own development...